solr/core/src/test/org/apache/solr/util/TestCborDataFormat.java

```diff
@@ -0,0 +1,221 @@
+  public void testRoundTrip() throws Exception {
+    String testCollection = "testRoundTrip";
+
+    MiniSolrCloudCluster cluster =
+        configureCluster(1)
+            .addConfig(
+                "conf", TEST_PATH().resolve("configsets").resolve("cloud-managed").resolve("conf"))
+            .configure();
+    try {
+      System.setProperty("managed.schema.mutable", "true");
+      CloudSolrClient client = cluster.getSolrClient();
+      CollectionAdminRequest.createCollection(testCollection, "conf", 1, 1).process(client);
+      modifySchema(testCollection, client);
+
+      byte[] b =
+          Files.readAllBytes(
+              new File(ExternalPaths.SOURCE_HOME, "example/films/films.json").toPath());
+      // every operation is performed twice. We should only take the second number
+      // so that we give JVM a chance to optimize that code
+      index(testCollection, client, createJsonReq(b), true);
+      index(testCollection, client, createJsonReq(b), true);
+
+      index(testCollection, client, createJavabinReq(b), true);
+      index(testCollection, client, createJavabinReq(b), true);
+
+      index(testCollection, client, createCborReq(b), true);
+      index(testCollection, client, createCborReq(b), false);
+
+      runQuery(testCollection, client, "javabin");
+      runQuery(testCollection, client, "javabin");
+      runQuery(testCollection, client, "json");
+      runQuery(testCollection, client, "json");
+      b = runQuery(testCollection, client, "cbor");
+      b = runQuery(testCollection, client, "cbor");
+      ObjectMapper objectMapper = new ObjectMapper(new CBORFactory());
+      Object o = objectMapper.readValue(b, Object.class);
+      List<Object> l = (List<Object>) Utils.getObjectByPath(o, false, "response/docs");
+      assertEquals(1100, l.size());
+    } finally {
+      System.clearProperty("managed.schema.mutable");
+      cluster.shutdown();
+    }
+  }
+
+  private void index(
+      String testCollection, CloudSolrClient client, GenericSolrRequest r, boolean del)
+      throws Exception {
+    RTimer timer = new RTimer();
+    client.request(r, testCollection);
+    System.out.println("INDEX_TIME: " + r.contentWriter.getContentType() + " : " + timer.getTime());
+    if (del) {
+      UpdateRequest req = new UpdateRequest().deleteByQuery("*:*");
+      req.setParam("commit", "true");
+      client.request(req, testCollection);
+    }
+  }
+
+  private byte[] runQuery(String testCollection, CloudSolrClient client, String wt)
+      throws SolrServerException, IOException {
+    NamedList<Object> result;
+    QueryRequest request;
+    RTimer timer = new RTimer();
+    request = new QueryRequest(new SolrQuery("*:*").setRows(1111));
+    request.setResponseParser(new InputStreamResponseParser(wt));
+    result = client.request(request, testCollection);
+    byte[] b = copyStream((InputStream) result.get("stream"));
+    System.out.println(wt + "_time : " + timer.getTime());
```

Comment:
[Q] So, as I read this code - `runQuery` will capture CBOR serialization in its perf numbers but **not** deserialization since we're not deserializing the received data on the client side.

Am I reading that right?  And if so, is that intentional?

[0] Also, doesn't forbiddenApis complain about the `System.out.println` usage here?  I swear that was one of the things we disallowed, though maybe we're looser about that in our tests...